zebra-chain/src/block/merkle.rs

```diff
@@ -78,6 +83,23 @@ fn hash(h1: &[u8; 32], h2: &[u8; 32]) -> [u8; 32] {
     w.finish()
 }
 
+/// Compute the root of a Merke tree as used in Bitcoin.
+/// `hashes` must contain the hashes of the tree leaves.
+/// The root is written to the the first element of the input vector.
+/// See [`Root`] for an important disclaimer.
+fn root(hashes: &mut Vec<[u8; 32]>) {
+    while hashes.len() > 1 {
+        *hashes = hashes
+            .chunks(2)
+            .map(|chunk| match chunk {
+                [h1, h2] => hash(h1, h2),
+                [h1] => hash(h1, h1),
+                _ => unreachable!("chunks(2)"),
+            })
+            .collect();
+    }
+}
```

Comment:
Do we need to check for duplicate auth digests here, like we do in the transaction ID merkle tree?

Reply:
I just checked the spec, and we pad with zeroes, rather than duplicate hashes:
https://zips.z.cash/zip-0244#block-header-changes